Kevin/standardize official ballot terminology

kshen0 opened the pull request:
## Overview
https://github.com/votingworks/vxsuite/issues/2783

Standardizes references for test/live mode from "Testing Mode", "Election Testing Mode", "Test Election Mode" etc to "Test Ballot Mode" and "Official Ballot Mode". Discussion in [Slack](https://votingworks.slack.com/archives/CEL6D3GAD/p1677281469337549).

## Demo Video or Screenshot

### VxCentralScan
![Screenshot 2023-02-28 at 6 44 21 PM](https://user-images.githubusercontent.com/1060688/222031883-ee8ebd8d-7a53-4734-82eb-a7dbf9cb53c3.png)
---
![Screenshot 2023-02-28 at 6 44 32 PM](https://user-images.githubusercontent.com/1060688/222031886-7e50c359-80f0-4d35-a7bd-19df60d9a5e6.png)
---
![Screenshot 2023-02-28 at 6 44 39 PM](https://user-images.githubusercontent.com/1060688/222031888-66461f17-4783-40c0-955f-99afec7c60aa.png)
---
![Screenshot 2023-02-28 at 6 46 06 PM](https://user-images.githubusercontent.com/1060688/222032015-779cc407-0273-4d00-8423-0c9fe6ff09f5.png)

---

### VxScan
![Screenshot 2023-02-24 at 4 49 13 PM](https://user-images.githubusercontent.com/1060688/221703773-1e4f1bc7-f5c3-42bc-a5e1-398ddaf79c2a.png)

---

### VxAdmin
![Screenshot 2023-02-27 at 2 59 06 PM](https://user-images.githubusercontent.com/1060688/221708546-ce9588f7-00d1-4f64-a13b-86bd6a656703.png)
---
![Screenshot 2023-02-27 at 3 01 10 PM](https://user-images.githubusercontent.com/1060688/221708887-5d943ab5-04b6-42c8-ab8f-0431c5a6018f.png)
---
![Screenshot 2023-02-28 at 6 43 50 PM](https://user-images.githubusercontent.com/1060688/222031939-769a1560-6d33-4b69-9fd2-8db1b206b507.png)


### VxMark
![Screenshot 2023-02-27 at 11 51 24 AM](https://user-images.githubusercontent.com/1060688/221708990-6d891f7f-caa3-46d8-bfca-981d9fd48a21.png)
---
![Screenshot 2023-02-27 at 4 10 45 PM](https://user-images.githubusercontent.com/1060688/221718365-0cdc0da6-0479-4c3c-b4da-7654c7eb02ba.png)


## Testing Plan
* Updated test expectations
* Manually rendered pages to confirm changes


## Checklist


- [x] I have added
      [logging](https://github.com/votingworks/vxsuite/tree/main/libs/logging)
      where appropriate to any new user actions, system updates such as file
      reads or storage writes, or errors introduced. **No new user actions or errors**
- [x] I have added JSDoc comments to any newly introduced exports **No new exports**
<!-- for user-facing changes, non-user facing changes can remove or ignore the below items -->
- [x] I have added a screenshot and/or video to this PR to demo the change
- [x] I have added the "user_facing_change" label to this PR to automate an
      announcement in #machine-product-updates

mattroe commented:
@kshen0 Overall LGTM

nit: why does the `Machine is in Test Ballot Mode` text wrap to two lines in one of the VxCentralScan screenshots, but not the other? Not the end of the world if it needs to, however it would be nice to be on one line.

Also, I think there might be one more place we use the `Live` copy that we discussed when originally thinking about the inconsistency, which is in the table of CVRs found on a USB drive: 

![image](https://user-images.githubusercontent.com/1476320/222018197-e49ee1c0-8494-43a6-a17d-c77defe3a3df.png)

I think changing the text from `Live` to `Official` in that table should work!

kshen0 commented:
@mattroe I think that modal is changed [here](https://github.com/votingworks/vxsuite/pull/3043/files#diff-04e11ccc303096875bee19db1c6ab09e279f01b2b9f0bb821111126af9bdd4a7R358) but I'll double check and post a screenshot

re: text wrapping, let me take a look with the correct screen resolution for VxCentralScan

mattroe commented:
@kshen0 sounds good, I didn't review all the file changes - was just looking at the screenshots

kshen0 commented:
Yup that's my bad, I forgot to add the screenshot. Thanks for flagging!

Updated the description:
* VxCentralScan: screenshots are now at 1920 x 1080 (banner has no wrap at this resolution)
* VxAdmin: Added screenshot for loading CVR files from USB

mattroe commented:
LGTM. I assume the VxScan font issue will be addressed once @kofi-q is finished with his design work.

adghayes commented on libs/utils/src/filenames.ts:
```diff
@@ -132,6 +132,9 @@ export function generateCastVoteRecordReportFilename(
 }
 
 /* Extract information about a CVR file from the filename */
+// Expected filename format with human-readable separators is:
+// [TEST__]machine_{machineId}__{numberOfBallots}_ballots__YYYY-MM-DD_HH-mm-ss.jsonl
+// This format is current as of 2023-02-22 and may be out of date if separator constants have changed
```
Thanks for updating this. Will be killing this exact format but will be using something similar.

adghayes commented on apps/central-scan/frontend/src/components/toggle_test_mode_button.test.tsx:
```diff
@@ -79,7 +79,45 @@ test('calls the callback on confirmation', () => {
-test('shows a modal when toggling to live mode', () => {
+test('toggle modal shows "official ballot mode" when toggling away from test ballot mode', () => {
+  const toggleTestMode = jest.fn();
+  const { getByText } = render(
+    <ToggleTestModeButton
+      canUnconfigure
+      isTestMode
+      isTogglingTestMode={false}
+      toggleTestMode={toggleTestMode}
+    />
+  );
+
+  // Click the button.
+  fireEvent.click(getByText('Toggle to Official Ballot Mode'));
+
+  getByText(
+    'Toggling to Official Ballot Mode will zero out your scanned ballots. Are you sure?'
+  );
```
Collectively we've been trending toward a couple recommended React testing library patterns:

- use `userEvent` rather than `fireEvent`
- use queries bound to the screen (e.g. `screen.getByText`) rather than queries bound to a specific render

These both simulate human interaction better and test better for accessibility. I've been meaning to write a doc summarizing these sorts of things.

It's not high priority - we're not putting any effort into old tests - but when I write new tests I try to follow these patterns. It's so common we're using the old pattern I don't think it's worth you making a change here unless you feel so inspired.

adghayes reviewed (approved):
LGTM! Just one comment that does not need to block you from merging. 

Thanks @kshen0 for diving in to the many screens and thanks @mattroe for doing a content review.

kshen0 commented:
Thanks @adghayes - updated to use testing best practices in [623a95d](https://github.com/votingworks/vxsuite/pull/3043/commits/623a95d31ff1a060acb0ba5cee534a5b3e8d49ca)
